superset-frontend/src/common/components/Modal.tsx

```diff
@@ -0,0 +1,125 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import styled from '@superset-ui/style';
+import { Modal as BaseModal } from 'src/common/components';
+import { t } from '@superset-ui/translation';
+import Button from 'src/views/CRUD/data/dataset/Button';
```

Comment:
Just a heads-up. I'm trying desperately to kill this Button implementation (and several others), and support its styles/features in the already-existing `src/components/Button`. For this PR, it'll be a trivial migration, but wanted to let you know in case this `src/views/CRUD/data/dataset/Button` component randomly disappears on you :)

Reply:
Good to know, I meant to use the one in `src/components`! I was also thinking of using the antd button.

Reply:
The `src/components` one will (soon!) support these bigger/wider buttons with a `cta` prop. I'm thinking of converting _that_ button to AntD (and have taken a couple steps toward that) but want to make sure it's used universally before I go messing with it again ;)

Reply:
Just checked, the Button in `src/views/CRUD...` above is actually using `src/common/components/Button` under the hood. So we could either move that styled button into its own file under `src/common/components` OR possibly add it as an optional style in the current `Button`? or I could move the styling into the new `Modal` component since the styling is only used in modals rn (see `AddDatasetModal`).

Reply:
Right... I'm moving it all into the current Button in https://github.com/apache/incubator-superset/pull/10649, so I can kill this component. I don't think you need to do any work on it, just wanted to keep you (and anyone else reading!) in the loop.